extension/encoding/awscloudwatchmetricstreamsencodingextension/opentelemetry_unmarshaler.go

```diff
@@ -4,15 +4,60 @@
 package awscloudwatchmetricstreamsencodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awscloudwatchmetricstreamsencodingextension"
 
 import (
+	"errors"
 	"fmt"
 
+	"github.com/gogo/protobuf/proto"
```

Comment:
Is it possible to use a different package that is not marked as deprecated?

Reply:
Good call. @constanca-m I think we can use https://pkg.go.dev/google.golang.org/protobuf@v1.36.5/encoding/protowire#ConsumeVarint, can you look into it please?

Reply:
Thank you, I have used the `encoding/binary` native package in the end.